### ISSUE TITLE
minimal-http-server-form-post (with SSL) - Callback was not called after client submiting form

### ISSUE THREAD

**Issue by trungnmh90** [warmcat/libwebsockets#2502]
`./minimal-http-server-form-post -s` and the index.html displayed well.
However, submitting form will cause my google-chrome returns error ERR_CONNECTION_CLOSED
debug shows `LWS_CALLBACK_HTTP` was not called when user clicked the send button.

Here is lws log
```
[2021/11/25 13:57:10:2568] U: LWS minimal http server POST | visit http://localhost:7681
[2021/11/25 13:57:10:2569] N: lws_create_context: LWS: 4.3.99-v4.3.0-96-g4c98a545, NET CLI SRV H1 H2 WS SS-JSON-POL ConMon IPv6-absent
[2021/11/25 13:57:10:2570] N: __lws_lc_tag:  ++ [wsi|0|pipe] (1)
[2021/11/25 13:57:10:2570] N: __lws_lc_tag:  ++ [vh|0|netlink] (1)
[2021/11/25 13:57:10:2582] N: __lws_lc_tag:  ++ [vh|1|default||7681] (2)
[2021/11/25 13:57:10:2582] N:  Vhost 'default' using TLS mode
[2021/11/25 13:57:10:2588] N: [vh|1|default||7681]: lws_socket_bind: source ads 0.0.0.0
[2021/11/25 13:57:10:2589] N: __lws_lc_tag:  ++ [wsi|1|listen|default||7681] (2)
[2021/11/25 13:57:15:5458] N: __lws_lc_tag:  ++ [wsisrv|0|default|(null)] (1)
[2021/11/25 13:57:15:5609] N: __lws_lc_untag:  -- [wsisrv|0|default|(null)] (0) 15.127ms
[2021/11/25 13:57:15:5622] N: __lws_lc_tag:  ++ [wsisrv|1|default|(null)] (1)
[2021/11/25 13:57:15:5695] N: __lws_lc_tag:  ++ [mux|0|default|h2_sid1_(wsisrv|1|default)] (1)
[2021/11/25 13:57:15:5712] N: __lws_lc_untag:  -- [mux|0|default|h2_sid1_(wsisrv|1|default)] (0) 1.682ms
[2021/11/25 13:57:15:6135] N: __lws_lc_tag:  ++ [mux|1|default|h2_sid3_(wsisrv|1|default)] (1)
[2021/11/25 13:57:15:6138] N: __lws_lc_untag:  -- [mux|1|default|h2_sid3_(wsisrv|1|default)] (0) 257μs
[2021/11/25 13:57:15:6157] N: __lws_lc_tag:  ++ [mux|2|default|h2_sid5_(wsisrv|1|default)] (1)
[2021/11/25 13:57:15:6160] N: __lws_lc_untag:  -- [mux|2|default|h2_sid5_(wsisrv|1|default)] (0) 266μs
[2021/11/25 13:57:15:7262] N: __lws_lc_tag:  ++ [mux|3|default|h2_sid7_(wsisrv|1|default)] (1)
[2021/11/25 13:57:15:7264] N: __lws_lc_untag:  -- [mux|3|default|h2_sid7_(wsisrv|1|default)] (0) 177μs
[2021/11/25 13:57:19:6474] N: __lws_lc_tag:  ++ [mux|4|default|h2_sid9_(wsisrv|1|default)] (1)
[2021/11/25 13:57:19:6476] N: __lws_lc_untag:  -- [mux|4|default|h2_sid9_(wsisrv|1|default)] (0) 134μs
[2021/11/25 13:57:19:6476] N: __lws_lc_untag:  -- [wsisrv|1|default|(null)] (0) 4.085s
[2021/11/25 13:57:19:6488] N: __lws_lc_tag:  ++ [wsisrv|2|default|(null)] (1)
[2021/11/25 13:57:19:6504] N: __lws_lc_untag:  -- [wsisrv|2|default|(null)] (0) 1.611ms
[2021/11/25 13:57:19:6521] N: __lws_lc_tag:  ++ [wsisrv|3|default|(null)] (1)
[2021/11/25 13:57:19:6545] N: __lws_lc_tag:  ++ [mux|5|default|h2_sid1_(wsisrv|3|default)] (1)
[2021/11/25 13:57:19:6547] N: __lws_lc_untag:  -- [mux|5|default|h2_sid1_(wsisrv|3|default)] (0) 197μs
[2021/11/25 13:57:19:6548] N: __lws_lc_untag:  -- [wsisrv|3|default|(null)] (0) 2.709ms
```
My OpenSSL package version, if it helps:
```
OpenSSL 1.1.1d  10 Sep 2019
```
Also, it is worth to mention that I test the case when client access the server over LAN network(not localhost)

**Comment by lws-team** [comment]
I think this is fixed in later main, eg, current head 44dd63a03dbea8dd22ed00824ae1450e9bffa4ae from a couple of days ago.

At least it works fine here on Firefox and Chromium.

**Comment by trungnmh90** [comment]
Silly me, you are right, my submodule of libwebsocket is untracked. Should have check that first. 
New release have fixed this issue.
